src/app/public/testing/phonefield/phone-field-fixture.ts

```diff
@@ -0,0 +1,143 @@
+    private skyTestId: string
+  ) {
+    this._debugEl = SkyAppTestUtility
+      .getDebugElementByTestId(fixture, skyTestId, 'sky-phone-field');
+
+    // The country selector needs extra time to initialize.
+    // Consumers shouldn't need to work around this so we do an extra detect here
+    fixture.detectChanges();
+  }
+
+  /**
+   * Sets the value of the input field for the phone field.
+   */
+  public async setInputText(inputText: string): Promise<any> {
+    const inputEl = this.phoneFieldInput;
+    inputEl.value = inputText;
+
+    SkyAppTestUtility.fireDomEvent(inputEl, 'input');
+    this.fixture.detectChanges();
+
+    SkyAppTestUtility.fireDomEvent(inputEl, 'change');
+
+    this.fixture.detectChanges();
+    return this.fixture.whenStable();
+  }
+
+  /**
+   * Opens the country selector, peforms a search, but makes no selection.
+   * @param searchText The name of the country to select.
+   */
+  public async searchCountry(searchText: string): Promise<NodeListOf<HTMLElement>> {
+    await this.openCountrySelection();
+
+    const countryFixture = await this.getCountryFixture();
+    const results = await countryFixture.search(searchText);
+
+    await this.waitForCountrySelection();
+    return results;
+  }
+
+  /**
+   * Opens the country selector, performs a search, and selects the first result (if any).
+   * @param searchText The name of the country to select.
+   */
+  public async selectCountry(searchText: string): Promise<any> {
+    await this.openCountrySelection();
+
+    const countryFixture = await this.getCountryFixture();
+    await countryFixture.searchAndSelectFirstResult(searchText);
+
+    return this.waitForCountrySelection();
+  }
+
+  //#region helpers
+
+  private get countryElement(): HTMLInputElement {
+    return this._debugEl.query(By.css('sky-country-field')).nativeElement;
+  }
+
+  private get countryFlagButton(): HTMLInputElement {
+    return this._debugEl.query(By.css('.sky-phone-field-country-btn .sky-btn')).nativeElement;
+  }
+
+  private get phoneFieldInput(): HTMLInputElement {
+    return this._debugEl.query(By.css('input[skyPhoneFieldInput]')).nativeElement;
+  }
+
+  private async getCountryFixture(): Promise<SkyCountryFieldFixture> {
+    if (this._countryFixture === undefined) {
+      // tag the country field with a sky test id
+      const countrySkyTestId = `${this.skyTestId}-country`;
+      this.setSkyTestId(this.countryElement, countrySkyTestId);
+
+      // initialize the country fixture
+      this._countryFixture = new SkyCountryFieldFixture(this.fixture, countrySkyTestId);
+
+      this.fixture.detectChanges();
+      await this.fixture.whenStable();
+    }
+
+    return this._countryFixture;
+  }
+
+  private async openCountrySelection(): Promise<any> {
+    this.countryFlagButton.click();
+    return this.waitForCountrySelection();
+  }
+
+  private setSkyTestId(element: HTMLElement, skyTestId: string) {
+    element.setAttribute('data-sky-id', skyTestId);
```

Comment:
We'll probably want to separate any logic that modifies the DOM to an adapter service. See `SkyPhoneFieldAdapterService` for an example of the pattern. Once you make that service, you can use Angular's `Renderer2` service's `setAttribute` method to modify attributes.

Reply:
I'll look at this.

Reply:
I think I see why this is desirable based on the docs, but I'm not seeing how I can inject it into a fixture since it's a concrete class and isn't instantiated via dependency injection.

Reply:
You should be able to inject it into the fixture's constructor like so:
```
constructor(
    private adapterService: SkyPhoneFieldFixtureAdapterService
) {}
```
You'll also need to make sure `SkyPhoneFieldFixtureAdapterService` is decorated as an `@Injectable`, and then you'll need to add it to your testing module's `providers` array. Let me know if you need help walking through it!